### PR TITLE
#3858 record all metrics periodically for performance tests

### PR DESCRIPTION
**Pull request by chia7712**
### Checklist
- [x] [Review Contribution Guidelines](https://ohara.readthedocs.io/en/latest/contributing.html)
- [x] Add Reviewers (see [Authors](https://github.com/oharastream/ohara#ohara-team))
- [x] Add Assignees (of course, you are the assignee by default!)
- [x] DON'T set a Milestone
- [x] DON'T add any labels (PR labels will be added automatically by Jenkins)
- [x] Link to an issue(You can do so by using the required by keyword: required by #3858)
- [x] Add unit tests (or please explain why this PR is not worth having new tests)
- [ ] Pass QA (Sometimes you may encounter an existing flaky test which would obstruct you from getting QA passed. If the failed tests are unrelated, you can add a comment in the PR thread)

### CODE REVIEW

**Comment by jackyoh** [comment]
build 錯誤, 很像是 maven repository 沒有找到 kafka 2.5.0 的 jar 檔

File: ohara-it/src/test/scala/com/island/ohara/it/performance/BasicTestPerformance.scala [hunk 1]
```diff
@@ -109,8 +116,80 @@ abstract class BasicTestPerformance extends WithRemoteWorkers {
...
+          fileWriter.write(line)
+          fileWriter.write("\n")
+      }
+    } finally Releasable.close(fileWriter)
+  }
+
   protected def sleepUntilEnd(): Long = {
-    TimeUnit.MILLISECONDS.sleep(durationOfPerformance.toMillis)
+    val end = CommonUtils.current() + durationOfPerformance.toMillis
+    while (CommonUtils.current() <= end) {
```
Comment (jackyoh) [inline ohara-it/src/test/scala/com/island/ohara/it/performance/BasicTestPerformance.scala]: 在執行 connector 時有可能會再另外寫入原始資料(setupInputData), 因此是否要在此區塊加入一個空的抽象化方法, 給實作使用

Comment (chia7712) [reply ohara-it/src/test/scala/com/island/ohara/it/performance/BasicTestPerformance.scala]: 收到

Comment (jackyoh) [reply ohara-it/src/test/scala/com/island/ohara/it/performance/BasicTestPerformance.scala]: Thank you.

**Comment by chia7712** [comment]
> build 錯誤, 很像是 maven repository 沒有找到 kafka 2.5.0 的 jar 檔

不小心提交了奇怪的東西 :(

**Comment by jackyoh** [comment]
> 不小心提交了奇怪的東西 :(

我猜是 mavenLocal()

**Pull request merged**